addons/sale/models/sale_order.py

```diff
@@ -1431,10 +1435,16 @@ def _get_invoiceable_lines(self, final=False):
                     # at the end of the invoice, in a specific dedicated section.
                     down_payment_line_ids.append(line.id)
                     continue
-                if pending_section:
-                    invoiceable_line_ids.append(pending_section.id)
-                    pending_section = None
-                invoiceable_line_ids.append(line.id)
+                if section_line_ids:
+                    if line.qty_to_invoice > 0 and not line.display_type:
+                        has_invoicable_product = True
+                    section_line_ids.append(line.id)
+                else:
+                    invoiceable_line_ids.append(line.id)
```

Comment:
```suggestion
                if section_line_ids:
                    if line.qty_to_invoice > 0 and not line.display_type:
                        has_invoicable_product = True
                    section_line_ids.append(line.id)
                    continue
                invoiceable_line_ids.append(line.id)
```

NIT (but if else is also good)

Reply:
Sure, will do that!
Thank you!